lib/charms/data_platform_libs/v0/object_storage.py

```diff
@@ -0,0 +1,262 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from collections import namedtuple
+from typing import Dict, List, Optional
+
+from charms.data_platform_libs.v0.data_interfaces import (
+    EventHandlers,
+    ProviderData,
+    RequirerData,
+    RequirerEventHandlers,
+)
+from ops import Model
+from ops.charm import (
+    CharmBase,
+    CharmEvents,
+    RelationBrokenEvent,
+    RelationChangedEvent,
+    RelationEvent,
+    RelationJoinedEvent,
+    SecretChangedEvent,
+)
+from ops.framework import EventSource, ObjectEvents
+from ops.model import Relation
+
+# The unique Charmhub library identifier, never change it
+LIBID = "fca396f6254246c9bfa5650000000000"
```

Comment:
TODO: these constants are to be updated later when we have a CharmHub space reserved for this charm.